plugins/modules/vmware_guest.py

```diff
@@ -3433,7 +3454,29 @@ def main():
         linked_clone=dict(type='bool', default=False),
         networks=dict(type='list', default=[], elements='dict'),
         resource_pool=dict(type='str'),
-        customization=dict(type='dict', default={}, no_log=True),
+        customization=dict(
+            type='dict',
+            default={},
+            options=dict(
+                autologon=dict(type='bool', default=False),
+                autologoncount=dict(type='int', default=1),
+                dns_servers=dict(type='list', elements='str'),
+                dns_suffix=dict(type='list', elements='str'),
+                domain=dict(type='str'),
+                domainadmin=dict(type='str'),
+                domainadminpassword=dict(type='str', no_log=True),
+                existing_vm=dict(type='bool'),
+                fullname=dict(type='str', default='Administrator'),
+                hostname=dict(type='str'),
+                hwclockUTC=dict(type='bool'),
+                joindomain=dict(type='str'),
+                joinworkgroup=dict(type='str', default='WORKGROUP'),
```

Comment:
@mariolenz thanks for working on this. Please add tests and changelog for this change.

Reply:
@Akasurde I'm afraid a changelog fragment and maybe one or two additional tests are not enough, there is a lot of code that I'll have to look at :-/ I'll reach out to you and the others when I've done that.